application.go

```diff
@@ -133,17 +134,32 @@ func (a Application) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
 
 	// Purge Workspace
 	a.Logger.Header("Removing source code")
-	cs, err := ioutil.ReadDir(a.ApplicationPath)
-	if err != nil {
-		return libcnb.Layer{}, fmt.Errorf("unable to list children of %s\n%w", a.ApplicationPath, err)
+	includeDirs, iset := a.ArtifactResolver.ConfigurationResolver.Resolve("BP_INCLUDE_FILES")
+	if includeDirs != "" {
+		if err := logic.Include(a.ApplicationPath, includeDirs); err != nil {
+			return libcnb.Layer{}, err
```

Comment:
```suggestion
			return libcnb.Layer{}, fmt.Errorf("unable to perform source-removal 'include' \n%w", err)
```
Just a minor suggestion - we tend to wrap all errors for easier tracing!